io/src/FileSystem/private/GoogleCloudStorageReadableFile.cpp

```diff
@@ -177,18 +178,18 @@ return arrow::Status::OK();
 		//}
 		//		results.GetResult().GetBody().read((char*)(out->get()->mutable_data()), bytesRead);
 
-		std::shared_ptr<arrow::ResizableBuffer> buffer;
-		AllocateResizableBuffer(arrow::default_memory_pool(), nbytes, &buffer);
+		arrow::Result<std::unique_ptr<arrow::ResizableBuffer>> buffer = AllocateResizableBuffer(nbytes, arrow::default_memory_pool());
 
-		results.read((char *) (buffer->mutable_data()), bytesRead);
+		results.read((char *) (buffer.ValueOrDie()->mutable_data()), bytesRead); 
```

Comment:
How we handle the "die" part? For example that arrow API says
`buffer.ValueOrDie()`
I think in most cases we will get the buffer but there will be be some cases where we get a "die" object/buffer ... how do we handle that? maybe we need to raise with low exception here cc @felipeblazing

Reply:
The exception handling of all of this code is in desperate need of cleanup. It has been outdated for a long time. I think that needs to be addressed by a different PR